app/src/main/java/com/google/android/gms/samples/wallet/CheckoutActivity.java

```diff
@@ -190,52 +194,45 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
      *     Data
      */
     private void handlePaymentSuccess(PaymentData paymentData) {
-        String token = paymentData.toJson();
+        String paymentInformation = paymentData.toJson();
 
         // Token will be null if PaymentDataRequest was not constructed using fromJson(String).
-        if (token != null) {
-            JSONObject tokenObject;
-            try {
-                tokenObject = new JSONObject(token);
-                // If the gateway is set to example, no payment information is returned - instead, the
-                // token will only consist of "examplePaymentMethodToken".
-                if (tokenObject
-                        .getJSONObject("paymentMethodData")
-                        .getJSONObject("tokenizationData")
-                        .getString("token")
-                        .equals("examplePaymentMethodToken")) {
-                    AlertDialog alertDialog =
-                            new AlertDialog.Builder(this)
-                                    .setTitle("Warning")
-                                    .setMessage(
-                                            "Gateway name set to \"example\" - please modify "
-                                                    + "Constants.java and replace it with your own gateway.")
-                                    .setPositiveButton("OK", null)
-                                    .create();
-                    alertDialog.show();
-                }
-
-                String billingName =
-                        tokenObject
-                                .getJSONObject("paymentMethodData")
-                                .getJSONObject("info")
-                                .getJSONObject("billingAddress")
-                                .getString("name");
-                Log.d("BillingName", billingName);
-                Toast.makeText(this, getString(R.string.payments_show_name, billingName), Toast.LENGTH_LONG)
-                        .show();
+        if (paymentInformation == null) {
+            return;
+        }
+        JSONObject paymentMethodData;
 
-                // Logging token string.
-                Log.d(
-                        "PaymentData",
-                        tokenObject
-                                .getJSONObject("paymentMethodData")
-                                .getJSONObject("tokenizationData")
-                                .getString("token"));
-            } catch (JSONException e) {
-                Log.e("handlePaymentSuccess failed", e.toString());
-                return;
+        try {
+            paymentMethodData = new JSONObject(paymentInformation).getJSONObject("paymentMethodData");
+            // If the gateway is set to example, no payment information is returned - instead, the
+            // token will only consist of "examplePaymentMethodToken".
+            if (paymentMethodData.getJSONObject("tokenizationData").getString("type").equals("PAYMENT_GATEWAY")
```

Comment:
consider reformatting with consistent newlines and indentation to make each paymentMethodData getter and comparison easier to read

if (paymentMethodData
        .getJSONObject("tokenizationData")
        .getString("type")
        .equals("PAYMENT_GATEWAY")
    && paymentMethodData
        .getJSONObject("tokenizationData")
        .getString("token")
        .equals("examplePaymentMethodToken")) {}